pkg/controller/util.go

```diff
@@ -326,3 +329,297 @@ func addToMap(m1, m2 map[string]string) map[string]string {
 	}
 	return m1
 }
+
+// returns the CloneRequest string which contains the pvc name (and namespace) from which we want to clone the image.
+func getCloneRequestPVC(pvc *v1.PersistentVolumeClaim) (string, error) {
+	cr, found := pvc.Annotations[AnnCloneRequest]
+	if !found || cr == "" {
+		verb := "empty"
+		if !found {
+			verb = "missing"
+		}
+		return cr, errors.Errorf("annotation %q in pvc \"%s/%s\" is %s\n", AnnCloneRequest, pvc.Namespace, pvc.Name, verb)
+	}
+	return cr, nil
+}
+
+func CreateCloneSourcePod(client kubernetes.Interface, image, verbose, pullPolicy, cr, secretName string, pvc *v1.PersistentVolumeClaim) (*v1.Pod, error) {
+	strArr := strings.Split(cr, "/")
+    var ns string = ""
+	if strArr == nil || len(strArr) < 2 {
+		glog.V(Vdebug).Infof("Wrong CloneRequest Annotation")
+		return nil, nil
+	}
+	ns = strArr[0]
+	pvcName := strArr[1]
+	
+	pod := MakeCloneSourcePodSpec(image, verbose, pullPolicy, cr, secretName, pvcName)
+
+	pod, err := client.CoreV1().Pods(ns).Create(pod)
+	if err != nil {
+		return nil, errors.Wrap(err, "source pod API create errored")
+	}
+	glog.V(Vuser).Infof("source pod \"%s/%s\" (image: %q) created\n", pod.Namespace, pod.Name, image)
+	return pod, nil
+}
+
+// return the clone source pod spec based on the target pvc.
+func MakeCloneSourcePodSpec(image, verbose, pullPolicy, cr, secret string, pvcName string) *v1.Pod {
+	// source pod name contains the pvc name
+	podName := fmt.Sprintf("%s-", CLONER_SOURCE_PODNAME)
+
+	pod := &v1.Pod{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		ObjectMeta: metav1.ObjectMeta{
+			GenerateName: podName,
+			Annotations: map[string]string{
+				AnnCreatedBy: "yes",
+			},
+		},
+		Spec: v1.PodSpec{
+			Affinity: &v1.Affinity{
+				NodeAffinity: &v1.NodeAffinity{
+					RequiredDuringSchedulingIgnoredDuringExecution: &v1.NodeSelector{
+						NodeSelectorTerms: []v1.NodeSelectorTerm{
+							{
+								MatchExpressions: []v1.NodeSelectorRequirement{
+									{
+										Key:      "kubevirt.io/Host-Assisted-Cloning",
+										Operator: v1.NodeSelectorOpIn,
+										Values: []string{"true"},
+									},
+								},
+							},
+						},
+	                },
+				},
+			},
+			Containers: []v1.Container{
+				{
+					Command:		[]string{"/bin/sh"},
+					Name:            CLONER_SOURCE_PODNAME,
+					Image:           image,
+					ImagePullPolicy: v1.PullPolicy(pullPolicy),
+					SecurityContext: &v1.SecurityContext{
+						Privileged: &[]bool{true}[0],
+						RunAsUser: &[]int64{0}[0],
+					},
+					VolumeMounts: []v1.VolumeMount{
+						{
+							Name:      ImagePathName,
+							MountPath: CLONER_IMAGE_PATH,
+						},
+						{
+							Name:      SocketPathName,
+							MountPath: CLONER_SOCKET_PATH,
+						},
+					},
+					Args: []string{"-c", "/tmp/script.sh source"},
```

Comment:
There's a const for this already: `CLONER_SCRIPT_ARGS` . Maybe not the best name but it saves _/tmp/script.sh_ from being hard-code as many times.

Reply:
Done.